### PR TITLE
rpc: add ValidateAddress grpc request

### LINKED ISSUE #842: Add validateaddress request to grpc

### PR DESCRIPTION
**Pull request by dnldd**
Fixes #842

### CODE REVIEW

**Comment by dnldd** [comment]
@jrick conflict resolved, should be good now.

**Comment by jrick** [comment]
@dnldd this has more conflicts, should be a quick review once that is resolved

File: rpc/documentation/api.md [hunk 1]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `string pub_key`: The hex encoded string public key associated with the address.
+
+- `string is_compressed`: True if address uses a compressed format, false if not.
+
+- `string account`:  The account name of the wallet owning the address.
+
+- `repeated string addresses`: A collection of addresses extracted from the address if it's a script.
+
+- `string hex`: The hex encoded redeem script if the address is p2sh
```
Comment (jrick) [inline rpc/documentation/api.md]: We should use `bytes` for binary data, not hex.

File: rpc/documentation/api.md [hunk 2]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `string script`: The script type
```
Comment (jrick) [inline rpc/documentation/api.md]: this should be an enum

File: rpc/documentation/api.md [hunk 3]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `string sigs_required`: The number of signatures required for the address.
```
Comment (jrick) [inline rpc/documentation/api.md]: don't use strings as numbers :)

this is just a doco issue, the actual message is using int32

File: rpc/documentation/api.md [hunk 4]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `bool is_watch_only`: True if the querying wallet is watch-only, false if not.
+
+- `string pub_key_addr`: The public key address.
+
+- `string pub_key`: The hex encoded string public key associated with the address.
+
+- `string is_compressed`: True if address uses a compressed format, false if not.
+
+- `string account`:  The account name of the wallet owning the address.
```
Comment (jrick) [inline rpc/documentation/api.md]: use the account number since that is what most of the rpcs use and return

File: rpc/documentation/api.md [hunk 5]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `bool is_mine`: True if the address is an address of querying wallet, false if not.
+
+- `bool is_watch_only`: True if the querying wallet is watch-only, false if not.
+
+- `string pub_key_addr`: The public key address.
+
+- `string pub_key`: The hex encoded string public key associated with the address.
+
+- `string is_compressed`: True if address uses a compressed format, false if not.
```
Comment (jrick) [inline rpc/documentation/api.md]: probably not necessary since we don't use uncompressed pubkeys anywhere

File: rpc/documentation/api.md [hunk 6]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+- `string address`: The validated address.
+
+- `bool is_mine`: True if the address is an address of querying wallet, false if not.
+
+- `bool is_watch_only`: True if the querying wallet is watch-only, false if not.
+
+- `string pub_key_addr`: The public key address.
+
+- `string pub_key`: The hex encoded string public key associated with the address.
```
Comment (jrick) [inline rpc/documentation/api.md]: Use `bytes`

File: rpc/documentation/api.md [hunk 7]
```diff
@@ -1559,6 +1560,42 @@ signature algorithms other than secp256k1.
...
+
+**Response:** `ValidateAddressResponse`
+
+- `bool is_valid`: True if valid, false if not.
+
+- `string address`: The validated address.
+
+- `bool is_mine`: True if the address is an address of querying wallet, false if not.
+
+- `bool is_watch_only`: True if the querying wallet is watch-only, false if not.
```
Comment (jrick) [inline rpc/documentation/api.md]: if this info is needed it should be moved out to a different rpc.  Imagine i wanted to check this, now i have to call ValidateAddress with some address from the wallet, but the address isn't actually needed to provide this info.

**Pull request merged**